File: CovidWatch/app/src/main/java/com/riskre/covidwatch/data/CovidWatchDatabase.java
package com.riskre.covidwatch.data;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

import java.util.UUID;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Database(entities = {ContactEvent.class}, version = 1)
public abstract class CovidWatchDatabase extends RoomDatabase {

    private static CovidWatchDatabase INSTANCE;

    public abstract ContactEventDAO contactEventDAO();

    private static final int NUMBER_OF_THREADS = 4;
    public static final ExecutorService databaseWriteExecutor =
            Executors.newFixedThreadPool(NUMBER_OF_THREADS);

    public static CovidWatchDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (CovidWatchDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            CovidWatchDatabase.class, "covidwatch_database")
                            .build();
                }
            }
        }
        return INSTANCE;
    }

}

File: CovidWatch/app/src/main/java/org/covidwatch/android/data/CovidWatchDatabase.kt
package org.covidwatch.android.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

@Database(entities = [ContactEvent::class], version = 1)
abstract class CovidWatchDatabase : RoomDatabase() {

    abstract fun contactEventDAO(): ContactEventDAO

    companion object {
        private const val NUMBER_OF_THREADS = 4
        val databaseWriteExecutor: ExecutorService =
            Executors.newFixedThreadPool(NUMBER_OF_THREADS)

        @Volatile
        private var INSTANCE: CovidWatchDatabase? = null

        fun getInstance(context: Context): CovidWatchDatabase =
            INSTANCE ?: synchronized(this) {
                INSTANCE ?: buildDatabase(context).also { INSTANCE = it }
            }

        private fun buildDatabase(context: Context) =
            Room.databaseBuilder(
                context.applicationContext,
                CovidWatchDatabase::class.java, "covidwatch.db"
            ).fallbackToDestructiveMigration().build()
    }
}